hyperopt/spark.py

```diff
@@ -468,15 +468,18 @@ def run_task_thread():
             params = self._get_spec_from_trial(trial)
 
             def run_task_on_executor(_):
+                import traceback
                 domain = base.Domain(
                     local_eval_function, local_space, pass_expr_memo_ctrl=None
                 )
 
                 try:
                     result = domain.evaluate(params, ctrl=None, attach_attachments=False)
                     yield result
-                except BaseException as e:
-                    yield e
+                except BaseException:
+                    error_string = traceback.format_exc()
+                    logger.error(error_string)
+                    yield BaseException(error_string)
```

Comment:
We may want to use `yield type(e)(error_string)` to preserve the error type instead of using `BaseException`.